File: src/day-029/get-how-many-duplicates.dart
int getHowManyDuplicates(str1, str2){
  if (str1.length > str2.length) return 0;
  final map = <String, List<int>>{};
  for (int i = 0; i < str1.length; i++){
    if(!map.containsKey(str1[i])){
      map[str1[i]] = [1,0];
    } else {
      map[str1[i]]?[0]++;
    }
  }
  for (int i = 0; i < str2.length; i++){
    if(map.containsKey(str2[i])){
      map[str2[i]]?[1]++;
    }
  }
  int min = double.maxFinite.toInt();
  map.forEach((_, value){
    var result = (value[1]/value[0]).floor();
    if (result < min){
      min = result;
    }    
  });
  return min;
}

File: src/60-days-01/day-029/get-how-many-duplicates.ts
function getHowManyDuplicates(str1: string, str2: string) {
  if (str1.length > str2.length) return 0;
  const map = new Map<string, Array<number>>();
  for (let i = 0; i < str1.length; i++) {
    const letter = str1[i];
    if (!map.has(letter)) {
      map.set(letter, [1, 0]);
    } else {
      const arr = map.get(letter);
      if (arr) {
        arr[0]++;
        map.set(letter, arr);
      }
    }
  }
  for (let i = 0; i < str2.length; i++) {
    const letter = str2[i];
    if (map.has(letter)) {
      const arr = map.get(letter);
      if (arr) {
        arr[1]++;
        map.set(letter, arr);
      }
    }
  }
  let min = Number.MAX_VALUE;
  map.forEach((value, key) => {
    let result: number = Math.floor(value[1] / value[0]);
    if (result < min) {
      min = result;
    }
  });
  return min;
}
